src/localplayer.cpp

```diff
@@ -512,12 +512,9 @@ void LocalPlayer::applyControl(float dtime)
 		}
 	}
 
-	if(continuous_forward)
-		speedH += move_direction;
-
-	if(control.up)
+	if(control.up || continuous_forward)
 	{
-		if(continuous_forward)
+		if(control.up && continuous_forward && fast_move)
```

Comment:
Shouldn't this be `control.up || continuous_forward && fast_move` (or just `fast_move` since the rest is checked above)?